bundles/AdminBundle/Resources/public/js/pimcore/settings/redirects.js

```diff
@@ -286,7 +286,15 @@ pimcore.settings.redirects = Class.create({
                 flex: 150, sortable:true, dataIndex: "expiry",
                 editor: {
                     xtype: 'datefield',
-                    format: 'Y-m-d'
+                    format: 'Y-m-d',
+                    listeners: {
+                        change: function (column, value, oldValue, eOpts) {
```

Comment:
Maybe there's something provided by the `DateField` that covers that as well? 
https://docs.sencha.com/extjs/7.0.0/classic/Ext.form.field.Date.html

Reply:
With the reset method we can restore the old valid value.